clients/cloud/java/src/main/java/io/confluent/examples/clients/ConsumerExample.java

```diff
@@ -42,23 +49,30 @@ public static void main(String[] args) throws Exception {
     Properties props = loadConfig(args[0]);
 
     // Add additional properties.
-    props.put("enable.auto.commit", "true");
-    props.put("auto.commit.interval.ms", "1000");
-    props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
-    props.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
-
-    // WARNING: These settings are for CLI tools / examples. Don't use in production unless you want to reset the consumer app every reboot.
-    props.put("group.id", "example" + new Random().nextInt());
-    props.put("auto.offset.reset", "earliest");
+    props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+    props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonDeserializer");
+    props.put(ConsumerConfig.GROUP_ID_CONFIG, "java_example_group_1");
+    props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
 
-    final Consumer<String, String> consumer = new KafkaConsumer<String, String>(props);
+    final Consumer<String, JsonNode> consumer = new KafkaConsumer<String, JsonNode>(props);
     consumer.subscribe(Arrays.asList(topic));
 
+    Long total_count = 0L;
+    String key;
```

Comment:
key and value should be in the `while (true)` loop (only scope where they are used)

Reply:
I declared them outside the while loop purposely to avoid instantiating a new object each time, but my understanding may be incorrect and if so, then I'm happy to move inside the loop.

Reply:
After some Googling: I suppose today's compilers are smart enough to handle this.  I'll move it in.